File: lib_internet/src/main/java/world/share/lib_internet/http/exception/ApiException.java
package world.share.lib_internet.http.exception;


/**
 * @author walle
 * 接口异常判断处理
 */
public class ApiException extends Exception {
    private int code;
    private String message;

    public ApiException(Throwable cause) {
        super(cause);
    }


    public ApiException(Throwable cause, int code, String message) {
        super(message,cause);
        this.code = code;
        this.message = message;
    }


    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    @Override
    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: lib_base/src/main/java/world/share/lib_base/internet/http/exception/ApiException.kt
package world.share.lib_base.internet.http.exception

import java.lang.Exception

/**
 * @author walle
 * 接口异常判断处理
 */
class ApiException : Exception {
    var code = 0
    override var message: String? = null

    constructor(cause: Throwable?) : super(cause) {}
    constructor(cause: Throwable?, code: Int, message: String?) : super(message, cause) {
        this.code = code
        this.message = message
    }

    @JvmName("setMessage1")
    fun setMessage(message: String?) {
        this.message = message
    }
}